hw3/01-characters.js

```diff
@@ -1,4 +1,75 @@
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
+const sectionToAddTo = document.querySelector('section');
+sectionToAddTo.classList.add('container-fluid', 'row');
 
-fetch(url);
+// fetch the data from the API
+fetch(url)
+  .then((data) => {
+    return data.json();
+  }) // return the json of the data
+  .then((characters) => {
+    displayCharacters(characters, sectionToAddTo);
+  }) // call fn to display it (append to the given element)
+  .catch((error) => {
+    console.log(error);
+  }) // catch any errors and throw them at the console
+  .finally(() => {
+    console.log('API call completed');
+  }); // why not have a finally? Sounds like a decent idea!
+
+// remove all the child elements for a given element
+function removeAllChildren(element) {
+  while (element.firstChild !== null) {
+    element.removeChild(element.firstChild);
+  }
+}
+
+function displayCharacters(characters, location) {
```

Comment:
This code could be broken into smaller components to make it cleaner

Reply:
I agree, but for the purposes of this assignment I will keep it in. I feel its reasonably clear enough for what it does